### PR TITLE
Cleanup futures 0.1 from jormungandr

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

File: jormungandr/Cargo.toml [hunk 1]
```diff
@@ -52,7 +52,7 @@ slog-scope = "4.1"
 slog-term = "2.4.0"
 structopt = "^0.3"
 thiserror = "1.0.16"
-tokio = { version = "0.2.12", features = ["signal"] }
+tokio = { version = "0.2.12", features = ["rt-threaded", "time", "sync", "stream", "signal", "test-util", "blocking"] }
```
Comment (mzabaluev) [inline jormungandr/Cargo.toml:55]: So many features are added with this small change, are these needed for other things to work?

Comment (mzabaluev) [reply jormungandr/Cargo.toml:55]: I guess it's leaner than "full" in any case...

File: jormungandr/src/blockchain/process.rs [hunk 2]
```diff
@@ -104,7 +104,7 @@ impl Process {
 
                 info!(logger, "receiving block from leadership service");
 
-                info.timeout_spawn_failable_std(
+                info.timeout_spawn_failable(
```
Comment (mzabaluev) [inline jormungandr/src/blockchain/process.rs:107]: Might as well rename to `*_fallible` which is the preferred Rust lingo ;)

File: jormungandr/src/log/stream.rs [hunk 3]
```diff
@@ -76,24 +82,25 @@ impl<'a, S: Debug, M: Debug> Debug for LoggingStream<'a, S, M> {
...
-    M: Display,
+    S: Stream + Unpin,
+    S::Item: Debug + Unpin,
+    M: Display + Unpin,
 {
     type Item = S::Item;
-    type Error = S::Error;
 
-    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
-        match try_ready!(self.stream.poll()) {
```
Comment (mzabaluev) [inline jormungandr/src/log/stream.rs:87]: How did this even work?

File: jormungandr/src/log/stream.rs [hunk 4]
```diff
@@ -76,24 +82,25 @@ impl<'a, S: Debug, M: Debug> Debug for LoggingStream<'a, S, M> {
 
 impl<'a, S, M> Stream for LoggingStream<'a, S, M>
 where
-    S: Stream,
-    S::Item: Debug,
-    M: Display,
+    S: Stream + Unpin,
+    S::Item: Debug + Unpin,
+    M: Display + Unpin,
```
Comment (mzabaluev) [inline jormungandr/src/log/stream.rs:87]: I hope this is acceptable sacrifice for being able to use `Pin::into_inner` for convenience. Not every stream is `Unpin`. On the other hand, we hardly use this adapter and we should transition to `tracing` as our logging framework in the future, so it's OK to just remove the whole thing.

Look into [pin-project](https://crates.io/crates/pin-project) as a nice way to do `Pin` projections.

File: jormungandr/src/utils/task.rs [hunk 5]
```diff
@@ -194,10 +194,10 @@ impl TokioServiceInfo {
         self.name
     }
 
-    /// Access the service's executor
+    /// Access the service's handle
     #[inline]
-    pub fn executor(&self) -> &TaskExecutor {
-        &self.executor
+    pub fn handle(&self) -> &Handle {
```
Comment (mzabaluev) [inline jormungandr/src/utils/task.rs:199]: Might name the member more descriptively, like `runtime_handle`.

File: jormungandr/src/log/stream.rs (hunk 3 above)
Comment (eugene-babichenko) [reply jormungandr/src/log/stream.rs:87]: black magic

File: jormungandr/src/log/stream.rs (hunk 4 above)
Comment (eugene-babichenko) [reply jormungandr/src/log/stream.rs:87]: Thanks, will try it!

Comment (eugene-babichenko) [reply jormungandr/src/log/stream.rs]: Worked well for logs. Not sure if will work at all for `FireForgetSchedulerFuture`, but will look how to simplify that one too.

**Pull request merged**